pkg/nodebootstrap/userdata.go

```diff
@@ -139,31 +140,40 @@ func makeKubeletExtraConf(kubeletExtraConf *api.InlineDocument) (cloudconfig.Fil
 }
 
 func makeBootstrapEnv(clusterConfig *api.ClusterConfig, np api.NodePool) cloudconfig.File {
-	variables := []string{
-		fmt.Sprintf("CLUSTER_NAME=%s", clusterConfig.Metadata.Name),
-		fmt.Sprintf("API_SERVER_URL=%s", clusterConfig.Status.Endpoint),
-		fmt.Sprintf("B64_CLUSTER_CA=%s", base64.StdEncoding.EncodeToString(clusterConfig.Status.CertificateAuthorityData)),
-		fmt.Sprintf("NODE_LABELS=%s", kvs(np.BaseNodeGroup().Labels)),
-		fmt.Sprintf("NODE_TAINTS=%s", utils.FormatTaints(np.NGTaints())),
+	ng := np.BaseNodeGroup()
+	variables := map[string]string{
+		"CLUSTER_NAME":   clusterConfig.Metadata.Name,
+		"API_SERVER_URL": clusterConfig.Status.Endpoint,
+		"B64_CLUSTER_CA": base64.StdEncoding.EncodeToString(clusterConfig.Status.CertificateAuthorityData),
+		"NODE_LABELS":    formatLabels(ng.Labels),
+		"NODE_TAINTS":    utils.FormatTaints(np.NGTaints()),
+	}
+
+	if ng.MaxPodsPerNode > 0 {
+		variables["MAX_PODS"] = strconv.Itoa(ng.MaxPodsPerNode)
 	}
```

Comment:
Can we add a unit test to check this ends up in userdatas?

Reply:
Done.